pandas/util/testing.py

```diff
@@ -685,8 +687,8 @@ def _check_types(l, r, obj='Index'):
             assert_class_equal(left, right, exact=exact, obj=obj)
             assert_attr_equal('dtype', l, r, obj=obj)
             # allow string-like to have different inferred_types
-            if l.inferred_type in ('string', 'unicode'):
-                assertIn(r.inferred_type, ('string', 'unicode'))
+            if l.inferred_type in ('string', 'unicode', 'mixed'):
```

Comment:
hmm, you sure about this? 'mixed' could be lots of things, even an integer & string will trigger.